libfreerdp/core/gateway/wst.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Websocket Transport
+ *
+ * Copyright 2023 Michael Saxl <mike@mwsys.mine.bz>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FREERDP_LIB_CORE_GATEWAY_WEBSOCKET_TRANSPORT_H
+#define FREERDP_LIB_CORE_GATEWAY_WEBSOCKET_TRANSPORT_H
+
+#include <winpr/wtypes.h>
+#include <winpr/stream.h>
+#include <winpr/collections.h>
+#include <winpr/interlocked.h>
+
+#include <freerdp/log.h>
+#include <freerdp/utils/ringbuffer.h>
+#include <freerdp/api.h>
+
+#include <freerdp/freerdp.h>
+#include <freerdp/types.h>
+#include <freerdp/settings.h>
+
+typedef struct rdp_wst rdpWst;
+
+#include "http.h"
```

Comment:
best clean up includes, only include in header what is required for the function declarations/type definitions

Reply:
I've chosen to include openssl/ssl.h for BIO*.
If you prefer can also change it to #include "../../crypto/tls.h"